test/test_autograd.py

```diff
@@ -780,16 +780,49 @@ def test_sparse_mm_backward(self):
         sparse = torch.sparse_coo_tensor(size, requires_grad=True)
         dense = torch.randn(size, requires_grad=True)
 
-        z = sparse.mm(dense)
-        with self.assertRaisesRegex(RuntimeError,
-                                    "calculating the gradient of a sparse Tensor argument to mm is not supported."):
-            z.sum().backward()
-
-        z = dense.addmm(sparse, dense)
-        with self.assertRaisesRegex(RuntimeError,
-                                    "calculating the gradient of a sparse Tensor argument to mm is not supported."):
-            z.sum().backward()
+        with self.assertRaisesRegex(
+                RuntimeError,
+                "The backward pass for this operation requires the 'mat1' tensor to be strided,"):
+            z = dense.addmm(sparse, dense)
+
+        mm_test_cases = [
+            # a requires grad, a is sparse, b requires grad, b is sparse, error message
+            (True, True, False, False, "The backward pass for this operation requires the 'self'"),
+            (False, False, True, True, "The backward pass for this operation requires the 'mat2'"),
+            (False, True, True, False, None),
```

Comment:
nit: any reason not to include all True, all False?
Also (true, true, false, true) that corresponds to the old test case if I read this correctly?

Reply:
All false wouldn't be testing any sparse inputs, and none of them would require grad, so I don't think there's a reason to test that case here.